js/src/edit-block/components/editor-provider.js

```diff
@@ -41,19 +36,25 @@ const EditorProvider = ( {
 	settings,
 	children,
 } ) => {
-	const { setupEditor } = useEditor();
 	const {
+		setEditedPost,
+		setupEditorState,
 		updatePostLock,
 		updateEditorSettings,
 	} = useDispatch( 'core/editor' );
 	// @ts-ignore type declaration not available
 	const { createWarningNotice } = useDispatch( noticesStore );
 
-	// Iniitialize the editor.
 	// Ideally this should be synced on each change and not just something you do once.
 	useLayoutEffect( () => {
 		updatePostLock( settings.postLock );
-		setupEditor( post );
+
+		if ( setEditedPost ) {
+			setEditedPost( post.type, post.id );
```

Comment:
Fixes a console error in WP `6.5`:

```
wp.data.dispatch( 'core/editor' ).setupEditorState is deprecated since version 6.5. Please use wp.data.dispatch( 'core/editor' ).setEditedPost instead.
```